src/app/portlets/dot-edit-page/layout/components/dot-theme-selector/dot-theme-selector.component.spec.ts

```diff
@@ -0,0 +1,158 @@
+    get(): Observable<DotTheme[]> {
+        return Observable.of(mockDotThemes);
+    }
+}
+
+fdescribe('DotThemeSelectorComponent', () => {
+    let component: DotThemeSelectorComponent;
+    let fixture: ComponentFixture<DotThemeSelectorComponent>;
+    let de: DebugElement;
+    const messageServiceMock = new MockDotMessageService({
+        'editpage.layout.theme.header': 'Header',
+        'editpage.layout.theme.search': 'Search',
+        'dot.common.apply': 'Apply',
+        'dot.common.cancel': 'Cancel'
+    });
+    const siteServiceMock = new SiteServiceMock();
+    let dialog;
+    let paginatorService: PaginatorService;
+    let dotThemesService: DotThemesService;
+
+    beforeEach(() => {
+        DOTTestBed.configureTestingModule({
+            declarations: [DotThemeSelectorComponent],
+            imports: [DataGridModule, SiteSelectorModule, BrowserAnimationsModule],
+            providers: [
+                {
+                    provide: DotThemesService,
+                    useClass: DotThemesServiceeMock
+                },
+                {
+                    provide: DotMessageService,
+                    useValue: messageServiceMock
+                },
+                { provide: SiteService, useValue: siteServiceMock },
+                PaginatorService
+            ]
+        });
+
+        fixture = DOTTestBed.createComponent(DotThemeSelectorComponent);
+        component = fixture.componentInstance;
+        de = fixture.debugElement;
+        dialog = de.query(By.css('p-dialog')).componentInstance;
+        component.value = mockDotThemes[0];
+        paginatorService = de.injector.get(PaginatorService);
+        dotThemesService = de.injector.get(DotThemesService);
+    });
+
+    describe('Dialog', () => {
+        beforeEach(() => {
+            fixture.detectChanges();
+        });
+
+        it('should be visible on init', () => {
+            expect(dialog.visible).toBeTruthy();
+        });
+
+        it('should emit close event when click on cancel button', () => {
+            const cancelBtn = de.query(By.css('.cancel')).nativeElement;
+            spyOn(component.close, 'emit');
+            cancelBtn.click();
+            expect(component.close.emit).toHaveBeenCalled();
+        });
+
+        it('should not be draggable, modal and have dismissable Mask', () => {
+            expect(dialog.closable).toBe(true, 'closable');
+            expect(dialog.draggable).toBe(false, 'draggable');
+            expect(dialog.modal).toBe(true, 'modal');
+        });
+
+        it('should call the apply method and emit the selected value', () => {
+            const applyBtn = de.query(By.css('.apply')).nativeElement;
+            spyOn(component, 'apply').and.callThrough();
+            spyOn(component.selected, 'emit');
+            component.current = mockDotThemes[1];
+            fixture.detectChanges();
+            applyBtn.click();
+
+            expect(component.apply).toHaveBeenCalledTimes(1);
```

Comment:
we don't need check this

Reply:
Because you are checking everything that the apply method does, so you don't need to check the call to the method itself.